packages/go_router_builder/lib/src/route_config.dart

```diff
@@ -325,8 +325,11 @@ RouteBase get $_routeGetterName => ${_routeDefinition()};
         : '''
 routes: [${_children.map((RouteConfig e) => '${e._routeDefinition()},').join()}],
 ''';
-    final String navigatorKey =
-        _key == null || _key!.isEmpty ? '' : 'navigatorKey: $_key,';
+    final String navigatorKeyParameterName =
+        _isShellRoute ? 'navigatorKey' : 'parentNavigatorKey';
+    final String navigatorKey = _key == null || _key!.isEmpty
+        ? ''
+        : '$navigatorKeyParameterName: $_key,';
```

Comment:
In order to test this, I was thinking to do the same as in `packages/go_router_builder/test/builder_test.dart` and use `@ShouldGenerate` to make sure we use the correct param name like so:

<details>
<summary>go_router_builder/test/test_inputs/_shell_test_input.dart</summary>

```dart
import 'package:flutter/src/widgets/framework.dart';
import 'package:flutter/src/widgets/navigator.dart';
import 'package:go_router/go_router.dart';
import 'package:source_gen_test/source_gen_test.dart';

@ShouldGenerate(r'''
RouteBase get $shellRouteWithKey => ShellRouteData.$route(
      factory: $ShellRouteWithKeyExtension._fromState,
      navigatorKey: ShellRouteWithKey.$navigatorKey,
      routes: [
        GoRouteData.$route(
          path: '/foo',
          factory: $GoRouteWithKeyExtension._fromState,
          parentNavigatorKey: GoRouteWithKey.$navigatorKey,
        ),
      ],
    );

extension $ShellRouteWithKeyExtension on ShellRouteWithKey {
  static ShellRouteWithKey _fromState(GoRouterState state) =>
      const ShellRouteWithKey();
}

extension $GoRouteWithKeyExtension on GoRouteWithKey {
  static GoRouteWithKey _fromState(GoRouterState state) =>
      const GoRouteWithKey();

  String get location => GoRouteData.$location(
        '/foo',
      );

  void go(BuildContext context) => context.go(location);

  void push(BuildContext context) => context.push(location);

  void pushReplacement(BuildContext context) =>
      context.pushReplacement(location);
}
''')
@TypedShellRoute<ShellRouteWithKey>(
  routes: <TypedRoute<RouteData>>[
    TypedGoRoute<GoRouteWithKey>(path: '/foo'),
  ],
)
class ShellRouteWithKey extends ShellRouteData {
  const ShellRouteWithKey();

  static final GlobalKey<NavigatorState> $navigatorKey =
      GlobalKey<NavigatorState>();
}

class GoRouteWithKey extends GoRouteData {
  const GoRouteWithKey();

  static GlobalKey<NavigatorState> get $navigatorKey => GlobalKey();
}
```
</details>

However, if I do this, I have a warning on the first 2 imports that `flutter` is not a dependency.
What do you prefer?
1. Ignore the warning
2. I add `flutter` to devDependencies
3. Else?

Reply:
Just add an example in the example folder, and add a test for that example to make sure everything works correctly. We also have a test to ensure the .g.dart is always the latest code. So if someone breaks the route_config.dart either the example test will fail or the .g.dart test will fail.